src/org/labkey/test/tests/SampleTypeLimitsTest.java

```diff
@@ -67,83 +71,155 @@ private void doSetup()
+        table.clickEditRow("1");
+        verifyInvalidLookupSample("quf_lookUpField", "Sample3", null);
+        verifyValidLookupSample("quf_lookUpField", "Sample2");
+
+        log("Verifying editing list row with the sample RowId");
+        table.clickEditRow("1");
+        SelectRowsCommand command = new SelectRowsCommand("samples", SAMPLE_TYPE_NAME);
+        command.setFilters(Arrays.asList(new Filter("Name", "Sample1")));
+        SelectRowsResponse response = command.execute(createDefaultConnection(), getProjectName());
+        verifyValidLookupSample("quf_lookUpField", response.getRows().get(0).get("RowId").toString(), "Sample1", "query", false);
+    }
+
+    private void verifyInvalidLookupSample(String fieldName, String sampleValue, @Nullable String expectedErrorMsg)
+    {
+        setFormElement(Locator.name(fieldName), sampleValue);
         clickButton("Submit");
 
         String errMsg = Locators.labkeyError.findElement(getDriver()).getText();
-        assertEquals("Expected error is different", "Could not convert value: Sample2", errMsg);
+        assertEquals("Expected error is different", expectedErrorMsg == null ? "Could not convert value: " + sampleValue : expectedErrorMsg, errMsg);
+    }
 
-        setFormElement(Locator.name("quf_lookUpField"), "Sample1");
+    private void verifyValidLookupSample(String fieldName, String sampleValue)
+    {
+        verifyValidLookupSample(fieldName, sampleValue, sampleValue, "query", false);
+    }
+
+    private void verifyValidLookupSample(String fieldName, String sampleValue, String sampleDisplay, String dataRegionName, boolean navigateViaBreadcrumb)
+    {
+        setFormElement(Locator.name(fieldName), sampleValue);
         clickButton("Submit");
 
+        if (navigateViaBreadcrumb)
+            clickAndWait(Locator.tagWithClass("ol", "breadcrumb").childTag("li").index(1).childTag("a"));
+
         log("Verifying row is inserted correctly");
-        table = new DataRegionTable("query", getDriver());
-        assertEquals("Lookup field value is incorrect", Arrays.asList("Sample1"), table.getColumnDataAsText("lookUpField"));
+        DataRegionTable table = DataRegionTable.DataRegion(getDriver()).withName(dataRegionName).waitFor();
+        assertEquals("Lookup field value is incorrect", sampleDisplay, table.getDataAsText(0, "lookUpField"));
+    }
+
+    private void insertSampleTypeRow(String regionName, String rowValue)
+    {
+        DataRegionTable table = DataRegionTable.DataRegion(getDriver()).withName(regionName).waitFor();
+        table.clickInsertNewRow();
+        setFormElement(Locator.name("quf_Name"), rowValue);
+        setFormElement(Locator.name("quf_label"), rowValue);
+        clickButton("Submit");
+    }
+
+    @Test
+    public void testDeriveSamplesLookupFields() throws IOException, CommandException
+    {
+        goToProjectHome();
+
+        log("Create sample type with lookup field to SAMPLE_TYPE_NAME");
```

Comment:
I think rather than use SAMPLE_TYPE_NAME you may want to say: 
`log("Create sample type with lookup field to " + SAMPLE_TYPE_NAME);`
or something similar that gets the value of SAMPLE_TYPE_NAME.

Reply:
Yup, good catch. Fixed.